docs/build/smart-contracts/example-contracts/bls-signature.mdx

```diff
@@ -0,0 +1,490 @@
+
+:::
+
+:::caution
+
+The example presented below is intended for demonstration purpose only
+
+- It has **not** undergone security auditing.
+- It is **not** safe for use in production environments.
+
+Implementing a production-safe signature scheme requires deep understanding of the underlying cryptography and security considerations. **Use this at your own risk**.
+
+:::
+
+## Run the Example
+
+First go through the [Setup] process to get your development environment configured, then clone the `v22.0.1` tag of `soroban-examples` repository:
+
+[setup]: ../getting-started/setup.mdx
+
+```
+git clone -b v22.0.1 https://github.com/stellar/soroban-examples
+```
+
+Or, skip the development environment setup and open this example in [Gitpod][oigp].
+
+To run the tests for the example, navigate to the `bls_signature` directory, and use `cargo test`.
+
+```
+cd bls_signature
+cargo test
+```
+
+You should see the output:
+
+```
+running 1 test
+test test::test ... ok
+```
+
+## Code
+
+```rust title="bls_signature/src/lib.rs"
+#[contract]
+pub struct IncrementContract;
+
+// `DST `is the domain separation tag, intended to keep hashing inputs of your
+// contract separate. Refer to section 3.1 in the [Hashing to Elliptic
+// Curves](https://datatracker.ietf.org/doc/html/rfc9380) on requirements of
+// DST.
+const DST: &str = "BLSSIG-V01-CS01-with-BLS12381G2_XMD:SHA-256_SSWU_RO_";
+
+#[derive(Clone)]
+#[contracttype]
+pub enum DataKey {
+    Owners,
+    Counter,
+    Dst,
+}
+
+#[contracterror]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum AccError {
+    InvalidSignature = 1,
+}
+
+#[contractimpl]
+impl IncrementContract {
+    pub fn init(env: Env, agg_pk: BytesN<96>) {
+        // Initialize the account contract essentials: the aggregated pubkey and
+        // the DST. Because the message to be signed (which is
+        // the hash of some call stack) is the same for all signers, we can
+        // simply aggregate all signers (adding up the G1 pubkeys) and store it.
+        env.storage().persistent().set(&DataKey::Owners, &agg_pk);
+        env.storage()
+            .instance()
+            .set(&DataKey::Dst, &Bytes::from_slice(&env, DST.as_bytes()));
+        // initialize the counter, i.e. the business logic this signer contract
+        // guards
+        env.storage().instance().set(&DataKey::Counter, &0_u32);
+    }
+
+    pub fn increment(env: Env) -> u32 {
+        env.current_contract_address().require_auth();
+        let mut count: u32 = env.storage().instance().get(&DataKey::Counter).unwrap_or(0);
+        count += 1;
+        env.storage().instance().set(&DataKey::Counter, &count);
+        count
+    }
+}
+
+#[contractimpl]
+impl CustomAccountInterface for IncrementContract {
```

Comment:
This is rather misleading, why is increment contract also an account? These should be two separate, unrelated contracts. I don't think the doc needs to contain the increment contract at all. The same actually goes for the example - tests don't even cover the increment logic - why overload the reader with non-relevant code?

Reply:
Sure, the increment example is just the bare minimal logic that's used as the basis for many existing examples, so I don't think it should hurt to include it, but I see your point.

Reply:
But it's completely redundant here, isn't it? You're not using it to demonstrate anything.